config/user-pam.m4

```diff
@@ -0,0 +1,37 @@
+AC_DEFUN([ZFS_AC_CONFIG_USER_PAM], [
+	AC_ARG_ENABLE([pam],
+		AS_HELP_STRING([--enable-pam],
+		[install pam_zfs_key module [[default: check]]]),
+		[enable_pam=$enableval],
+		[enable_pam=check])
+
+	AC_ARG_WITH(pammoduledir,
+		AS_HELP_STRING([--with-pammoduledir=DIR],
+		[install pam module in dir [[$libdir/security]]]),
+		[pammoduledir="$withval"],[pammoduledir=$libdir/security])
+
+	AC_ARG_WITH(pamconfigsdir,
+		AS_HELP_STRING([--with-pamconfigsdir=DIR],
+		[install pam-config files in dir [[/usr/share/pamconfigs]]]),
+		[pamconfigsdir="$withval"],[pamconfigsdir=/usr/share/pam-configs])
+
+	AS_IF([test "x$enable_pam" != "xno"], [
+		AC_CHECK_HEADERS([security/pam_modules.h], [
+			enable_pam=yes
+		], [
+			AS_IF([test "x$enable_pam" == "xyes"], [
+				AC_MSG_FAILURE([
+	*** security/pam_modules.h missing, libpam0g-dev package required
+				])
+			],[
+				enable_pam=no
+			])
+		])
+	])
+	AS_IF([test "x$enable_pam" == "xyes"], [
+		DEFINE_PAM='--with "pam" --define "_pamconfigsdir $(pamconfigsdir)"'
```

Comment:
The reason, why I removed pammoduledir here is that is needs to be set dependent on libdir, so the pam module gets installed into `%{_libdir}/security`. The "correct" libdir is never determined in the "outer" automake (which generates the rpm package), but only set in the inner automake. Passing `pammoduledir` would prevent the correct discovery.

Reply:
So? You can still have both autodiscovery and overrides.
```
AC_ARG_WITH([pammoduledir],
    [AS_HELP_STRING([--with-pammoduledir], [Blahtext]),
    [pammoduledir="$withval"]
    [pammoduledir='${libdir}/security'; any_other_default_code;])
```

Reply:
No, I meant the other direction. If I forward the pamdir here to the rpm, the rpm will get a "wrong" pamdir of `/lib/security` instead of `/lib/x86_64-linux-gnu/security` forcibly set in any case, which in turn will always set the `--with-pammoduledir` config option for the "inner" zfs build. So the "inner" zfs build will always get a wrong pammoduledir. The root cause is probably that `libdir` is not discovered by automake, but by the rpmbuild in the middle, so the "outer" automake does never get the real libdir.

Reply:
I see.  Thanks for the clarification.